kernel/src/hil/flash.rs

```diff
@@ -0,0 +1,28 @@
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Error {
+}
+
+/// A block of writable persistent flash memory.
+pub trait Flash {
+    /// Integer type the size of a word.
+    type Word;
+
+    /// Set the client for this flash peripheral. The client will be called
+    /// when operations complete.
+    fn set_client(&self, client: &'static Client);
+
+    /// Read data
+    fn read(&self, offset: usize, buf: &mut [Self::Word]);
+
+    /// Write data
+    fn write(&self, offset: usize, buf: &mut [Self::Word]);
```

Comment:
I believe that this needs to be a&'static mut in order to allow the implementation to share the buffer with the caller (otherwise, the caller passes in a reference and never gets it back so the borrow checker can't tell when it's safe to mutate it again)